### ISSUE TITLE
ConCom Module - Years Served Tracking

### ISSUE THREAD

**Issue by tskeeley** [CON-In-A-Box/CIAB-Signin#16]
In the ConCom Module, on the main page currently, there is non-displayed code for years served.  This tracking needs to be cleaned up and updated (and moved out of a custom field and into a custom object (separate DB Table, see volunteer hour tracking for example)

Finished support - Need to display "was on concom" info for previous years
Function need - "I worked that year, but I'm not listed.  I need to request this to be added"
Admin list need - "Show me all users that have worked on concom for 10/20 years"
Function need - When someone is added to the ConCom, the "years worked" is updated to include the current con year.  Note:  If they are removed, the "years worked" does NOT change
Potential data - track what departments worked on what year

**Comment by AricStewart** [comment]
I am conformable with the PHP and Javascript and all that. I am less conformable changing Neon tables and such. 

If someone else is willing to do the Neon parts I am happy to update the web code to make use of it.

**Comment by AricStewart** [comment]
Currently waiting on Issue #27

**Comment by tskeeley** [comment]
#27 complete.  can this move back to active (when possible, low priority)

**Comment by AricStewart** [comment]
Addressed in pull request #71

**Comment by tskeeley** [comment]
still pending the following parts:

Function need - "I worked that year, but I'm not listed. I need to request this to be added"
Admin list need - "Show me all users that have worked on concom for 10/20 years"
Function need - When someone is added to the ConCom, the "years worked" is updated to include the current con year. Note: If they are removed, the "years worked" does NOT change
Potential data - track what departments worked on what year

PS:  do you want me to convert this to a project and individual issues? (which would be appropriate LOL)

**Comment by AricStewart** [comment]
> - Function need - "I worked that year, but I'm not listed. I need to request this to be added"


What should this do?  Send an e-mail? Put something in the database?

> - Admin list need - "Show me all users that have worked on concom for 10/20 years"

What page do we want this on? somewhere on the concom list?

> - Function need - When someone is added to the ConCom, the "years worked" is updated to include the current con year. Note: If they are removed, the "years worked" does NOT change

Added via the ConCom list? Ok, the removed but still show the list is tougher because at present I believe we fully delete the row.  This is going to be hard because there is no way to mark as "having" worked for the present year, but not presently working.

> - Potential data - track what departments worked on what year

That will be kept in the list so we already do this.

**Comment by tskeeley** [comment]
>  I worked that year...
short term - send email to admin.  Long term, enter a "to be approved" entry into the db for the division director.  That's a 2.0 thing at this point.

> Show me all users who....
Originally, I thought the admin tab.  We COULD put it on the vol-admin page (volunteers) which would make a lot of sense as it's part of the volunteer module.  It is NOT a public-facing function.

> removed from concom...
So, it sounds like there needs to be an "inactive" boolean field in the concom list, OR a seperate table of history.  your choice.  Let me know and I'll make it so.

> potential...
Fair point!  thanks!

**Comment by seanbseanb** [comment]
What's our prioritization for this issue?  Feels low...

**Comment by tskeeley** [comment]
Actually, this has to be high.  Since The "years worked" is showing on the main page again, when that goes live, there are people that will go crazy if the years they've worked aren't listed.  Also, since this is year 20 of Con, there is a critical need to know when people get to 20 years worked for recognition.  While the driving force behind this is ego, it will save us a lot fo time to have these "simple" buttons available before the concom sees them.

**Comment by AricStewart** [comment]
There is an events worked thing already present. How would I populate past data? It is all in Neon? Somewhere else?

**Comment by AricStewart** [comment]
Ok, I propose we remove this from this milestone and move it to futures. If having in inaccurate display is the issue we can disable that portion of the code until things are better here. Easy enough to do.

Thoughts? Feedback? Bueller?

**Comment by tskeeley** [comment]
Agreed.  disable away!

**Comment by AricStewart** [comment]
Done, Disable for 2018, we can expand it for 2019.

**Comment by tskeeley** [comment]
This also seemed related to #178
